src/main/java/org/gridsuite/geodata/server/GeoDataService.java

```diff
@@ -515,17 +515,25 @@ List<LineGeoData> getLinesByCountries(Network network, Set<Country> countries) {
 
         // we also want the destination substation (so we add the neighbouring country)
         Set<Country> countryAndNextTo = mapSubstationsByLine.entrySet().stream().flatMap(entry ->
-             Stream.of(entry.getValue().getLeft(), entry.getValue().getRight()).map(Substation::getNullableCountry).filter(Objects::nonNull)).collect(Collectors.toSet());
+             Stream.of(entry.getValue().getLeft(), entry.getValue().getRight())
+                     .filter(Objects::nonNull)
+                     .map(Substation::getNullableCountry)
+                     .filter(Objects::nonNull)).collect(Collectors.toSet());
 
         Map<String, SubstationGeoData> substationGeoDataDb = getSubstationMapByCountries(network, countryAndNextTo);
         List<LineGeoData> geoData = new ArrayList<>();
 
-        mapSubstationsByLine.forEach((key, value) -> {
-            LineGeoData geo = getLineGeoDataWithEndSubstations(linesGeoDataDb, substationGeoDataDb, key, value.getLeft(), value.getRight());
-            if (geo != null) {
-                geoData.add(geo);
-            }
-        });
+        mapSubstationsByLine.entrySet().stream()
+                .filter(entry -> entry.getValue() != null)
+                .filter(entry -> entry.getValue().getLeft() != null)
+                .filter(entry -> entry.getValue().getRight() != null)
```

Comment:
Only one filter call instead of 3 ?

Reply:
I see it clearer that way. Do you think it could have a noticeable influence on performance ?